From a7f807349dcc3297473d8ddc031206b2912f7301 Mon Sep 17 00:00:00 2001
From: Dario Binacchi <dario.binacchi@amarulasolutions.com>
Date: Tue, 17 Sep 2024 21:48:01 +0200
Subject: [PATCH] Fix compilation error on linux

This patch fixes the following compilation error raised by the bump to
version 1.13.1 in Buildroot:

net.c: In function 'babel_send':
net.c:199:27: error: 'IPV6_DONTFRAG' undeclared (first use in this function)
  199 |         cmsg->cmsg_type = IPV6_DONTFRAG;;

Signed-off-by: Dario Binacchi <dario.binacchi@amarulasolutions.com>
Upstream: https://github.com/jech/babeld/pull/121
---
 net.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/net.c b/net.c
index 7a7b57844e6c..822112299fd8 100644
--- a/net.c
+++ b/net.c
@@ -37,6 +37,10 @@ THE SOFTWARE.
 #include <arpa/inet.h>
 #include <errno.h>
 
+#if defined(__linux)
+#include <linux/in6.h>
+#endif
+
 #include "babeld.h"
 #include "util.h"
 #include "net.h"
-- 
2.43.0

